### DISCUSSION TITLE
Does Qodana 2022.2 support Java 17?

### DISCUSSION THREAD

**Discussion by vplagov** [JetBrains/Qodana#120] [Q&A] [answered]
[In my project](https://github.com/plagov/project-euler/blob/master/build.gradle.kts), I set Java 17 via Gradle's Toolchain. When I run the Qodana scan, it fails with the following error:
```
Preparing for the Project configuration stage ...
2022-08-22 07:11:16,555 [  22051]   WARN - #c.i.o.p.i.UnknownSdkInspectionCommandLineConfigurator - Failed to resolve JDK "openjdk-17": JDK "openjdk-17" is missing

The Project configuration stage completed in 1s 20ms
Please, specify sdk 'openjdk-17' for module 'project-euler.test'
2022-08-22 07:11:18,812 [  24308] SEVERE - #o.j.p.s.i.r.QodanaInspectionApplication - Cannot configure project to run inspections
```
I don't quite understand what should I configure and why?
Complete log available here - https://github.com/plagov/project-euler/runs/7946347098?check_suite_focus=true#step:3:155

Also, a second question.
In my project, I have the [Qodana configuration file](https://github.com/plagov/project-euler/blob/master/qodana.yaml) defined, but still, from the logs (same run as above), I see that Qodana couldn't pick it up. Did I misconfigure something in my setup?

**Comment by brichbash** [comment, marked as answer]
Hello,
Could you please try changing `openjdk-17` to `17` in [.idea/misc.xml]( https://github.com/plagov/project-euler/blob/master/.idea/misc.xml)? `openjdk-17` has been removed from [JetBrains CDN](https://download-cdn.jetbrains.com/jdk/feed/v1/jdks.json), so Qodana can't resolve it.
> Also, a second question.
In my project, I have the [Qodana configuration file](https://github.com/plagov/project-euler/blob/master/qodana.yaml) defined, but still, from the logs (same run as above), I see that Qodana couldn't pick it up. Did I misconfigure something in my setup?

Could you please clarify what is the problem? I see that you set `qodana.recommended` profile in `qodana.yaml`. And in the logs there is:
```
(Project opening) done.
The Project opening stage completed in 14s 410ms
Initializing project...Inspecting with the 'qodana.recommended' profile
```

**Reply by vplagov** [reply]
Thanks! I forgot I checked in the `.idea/misc.xml` for Qodana. Works now after the suggested change.

As for the second question, I was confused by these log entries at a very beginning of the log file:
```
!  No valid qodana.yaml found. Have you run qodana init? Running that for you...
Scanning project...
```
